lib/Directory.rakumod

```diff
@@ -54,6 +57,64 @@ class Directory {
         $out ~= "\nContains: $dircnt $subdir, $filecnt $file\n\n";
     }
 
+    #| Trivial pluralisation.
+    #| 1, 'file'                      --> "1 file"
+    #| 2, 'file'                      --> "2 files"
+    #| 3, 'directory', 'directories'  --> "3 directories"
+    #| 1, 'directory', 'directories'  --> "1 directory"
+    #| 0, 'directory', 'directories'  --> "0 directories"
+    sub _plural ( $count, $descr, $plural = '' --> Str ) {
+        $count
+            ~ ' '
+            ~ (    $count == 1 ?? $descr
+                !! $plural     ?? $plural
+                !! $descr ~ 's' );
+    }
+
+    # Alternate to .gist (usually a short one-line Str)
+    method Str ( --> Str ) {
+        my @str = "Directory:", $!dirpath.Str;
+        if self.exists {
+            @str.append( _plural( self.file-count, 'file') )
+                if self.file-count > 0;
+            @str.append( _plural( self.dir-count, 'directory', 'directories' ) )
+                if self.dir-count > 0;
+        }
+        else {
+            @str.append('(non-existant)');
+        }
+        @str.join(' ');
+    }
+
+    #| only count content entries once
```

Comment:
What if the number of directories/files change between now and the last time they were counted?

Reply:
Yes - maybe even an option `:cache` to only check once, and `:cache(False)` to not cache. I only did it that way for efficiency - but if you expect folks to hold on to instances of Directory then maybe even make `False` the default.